src/main/java/com/sproutsocial/nsq/RoundRobinPublisher.java

```diff
@@ -0,0 +1,224 @@
+                if (parent.isStopping) {
+                    throw new NSQException("publisher stopped");
+                }
+                return connectAttempt();
+            } else if (isFailover && canAttemptRecovery()) {
+                isFailover = false;
+                connectAttempt();
+                logger.info("using primary nsqd");
+                return true;
+            }
+            return !(isFailover || con == null);
+        }
+
+        private boolean canAttemptRecovery() {
+            return Util.clock() - failoverStart > failoverDurationSecs * 1000;
+        }
+
+        private boolean connectAttempt() {
+            if (con != null) {
+                con.close();
+            }
+            con = new PubConnection(client, this.hostAndPort, parent);
+            try {
+                con.connect(config);
+            } catch (IOException e) {
+                con.close();
+                con = null;
+                isFailover = true;
+                failoverStart = Util.clock();
+                logger.warn("Failed to connect to {}, will retry later", hostAndPort);
+                return false;
+            }
+            logger.info("publisher connected:{}", hostAndPort);
+            return true;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            ConnectionDetails that = (ConnectionDetails) o;
+            return hostAndPort.equals(that.hostAndPort);
+        }
+
+        @Override
+        public int hashCode() {
+            return hostAndPort.hashCode();
+        }
+
+
+        @Override
+        public String toString() {
+            return "ConnectionDetails{" +
+                    "hostAndPort=" + hostAndPort +
+                    '}';
+        }
+
+        protected synchronized void markFailure() {
+            con = null;
+            isFailover = true;
+            failoverStart = Util.clock();
+            logger.warn("Failed to connect to {}, will retry later", hostAndPort);
+        }
+
+        public void close() {
+            Util.closeQuietly(con);
+            con = null;
+        }
+    }
+
+    private final List<ConnectionDetails> daemonList;
+    private int nextDaemonIndex = 0;
+    private int failoverDurationSecs = 300;
+
+    private static final Logger logger = LoggerFactory.getLogger(RoundRobinPublisher.class);
+
+    public RoundRobinPublisher(Client client, String nsqd, String failoverNsqd, Publisher parent) {
+        super(client);
+        this.parent = parent;
+        Set<ConnectionDetails> connectionDetails = new HashSet<ConnectionDetails>();
+        daemonList = Collections.unmodifiableList(getConnectionDetails(nsqd, failoverNsqd, connectionDetails));
+    }
+
+    private List<ConnectionDetails> getConnectionDetails(String nsqd, String failoverNsqd, Set<ConnectionDetails> connectionDetails) {
```

Comment:
any reason not to make `connectionDetails` a variable scoped to this function only?